libfoundation/src/foundation-filters.cpp

```diff
@@ -340,19 +340,22 @@ static const char *url_table[256] =
 
 bool MCFiltersUrlEncode(MCStringRef p_source, MCStringRef& r_result)
 {
-    char *t_utf8_string;
-    char *s;
-    uint32_t l;
-    int32_t size;
+	MCAutoArray<char> t_utf8_string;
+	const char *s;
+	uindex_t l;
+	uindex_t size;
     
     // SN-2014-11-13: [[ Bug 14015 ]] We don't want to nativise the string,
     // but rather to encode it in UTF-8 and write the bytes (a '%' will be added).
-    if (!MCStringConvertToUTF8(p_source, t_utf8_string, l))
+	if (!MCStringConvertToUTF8(p_source,
```

Comment:
I'm pretty sure we have an MCAutoStringRefAsUTF8String class which makes this is a little neater.